src/poetry/repositories/http_repository.py

```diff
@@ -79,10 +79,13 @@ def _get_info_from_wheel(self, url: str) -> PackageInfo:
         from poetry.inspection.info import PackageInfo
 
         wheel_name = urllib.parse.urlparse(url).path.rsplit("/")[-1]
-        self._log(f"Downloading wheel: {wheel_name}", level="debug")
 
         filename = os.path.basename(wheel_name)
+        filepath = self._authenticator.get_cached_file_for_url(url)
+        if filepath:
+            return PackageInfo.from_wheel(filepath)
```

Comment:
We should apply the same optimization to `_get_info_from_sdist()`, probably even everywhere where `_download()` is used (i.e. also in `_links_to_data()`), shouldn't we?

Reply:
Thanks for the suggestion, I should have surely detected that myself 😅

I took the liberty to refactor a bit the code to avoid lots of duplications, which ended up in a nice context manager and code easier to follow IMO. Looking forward to see this performance improvement in... 1.5.0? 👀

Reply:
Further exploring the `PackageInfo.from_*` helper methods it looks to me like we could just invoke `PackageInfo.from_path` for both wheel and sdist (`DirectOrigin` already does that), thus removing the need of having two separate `_get_info_from_wheel` and `_get_info_from_sdist`. I'll let you decide on this

Reply:
`PackageInfo.from_wheel()` returns an empty `PackageInfo` if there is a `ValueError`, `PackageInfo.from_path()` will probably not. I don't know what the implications of that are. Thus, maybe keep it as is for now...